### PR TITLE
feat: [CDS-44348]: migrate artifact streams

### PR DESCRIPTION
**Pull request by prabu-kr**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38555)

### CODE REVIEW

File: 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/ACRArtifactStreamMapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @Override
+  public PrimaryArtifact getArtifactDetails(MigrationInputDTO inputDTO, Map<CgEntityId, CgEntityNode> entities,
+      Map<CgEntityId, Set<CgEntityId>> graph, ArtifactStream artifactStream,
+      Map<CgEntityId, NGYamlFile> migratedEntities) {
+    AcrArtifactStream acrArtifactStream = (AcrArtifactStream) artifactStream;
+    NgEntityDetail connector =
+        migratedEntities.get(CgEntityId.builder().type(CONNECTOR).id(acrArtifactStream.getSettingId()).build())
+            .getNgEntityDetail();
+    return PrimaryArtifact.builder()
+        .sourceType(ArtifactSourceType.JENKINS)
```
Comment (puthrayaharness) [inline 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/ACRArtifactStreamMapper.java:41]: Please fix.

File: 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/AMIArtifactStreamMapper.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import software.wings.beans.artifact.AmiArtifactStream;
+import software.wings.beans.artifact.ArtifactStream;
+import software.wings.ngmigration.CgEntityId;
+import software.wings.ngmigration.CgEntityNode;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class AMIArtifactStreamMapper implements ArtifactStreamMapper {
```
Comment (puthrayaharness) [inline 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/AMIArtifactStreamMapper.java:32]: Please do not map AMI yet. This is not supported & it is still in develop

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/AmazonS3ArtifactStreamMapper.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    NgEntityDetail connector =
+        migratedEntities.get(CgEntityId.builder().type(CONNECTOR).id(amazonS3ArtifactStream.getSettingId()).build())
+            .getNgEntityDetail();
+    return PrimaryArtifact.builder()
+        .sourceType(ArtifactSourceType.AMAZONS3)
+        .spec(AmazonS3ArtifactConfig.builder()
+                  .primaryArtifact(true)
+                  .connectorRef(ParameterField.createValueField(MigratorUtility.getIdentifierWithScope(connector)))
+                  .bucketName(ParameterField.createValueField(amazonS3ArtifactStream.getJobname()))
+                  .filePath(amazonS3ArtifactStream.getArtifactPaths() != null
```
[the remark this answers is not in the document]
Comment (prabu-kr) [reply 100-migrator/src/main/java/io/harness/ngmigration/service/artifactstream/AmazonS3ArtifactStreamMapper.java]: Done

**Comment by AnshuH** [comment]
trigger ti1

**Comment by prabu-kr** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]

**Pull request merged**